src/client/http.rs

```diff
@@ -21,16 +21,16 @@ pub struct Options {
     pub chunk_size: Option<u16>
 }
 
-pub struct HttpClient<'a> {
-    credentials: Credentials<'a>,
+pub struct HttpClient {
+    credentials: Credentials,
     serializer: Box<Serializer + Send + Sync>,
     hurl: Box<Hurl + Send + Sync>,
-    hosts: Vec<&'a str>,
+    hosts: Vec<String>,
     pub max_batch: u16
 }
 
-impl<'a> HttpClient<'a> {
-    pub fn new(credentials: Credentials<'a>, serializer: Box<Serializer + Send + Sync>, hurl: Box<Hurl + Send + Sync>) -> HttpClient<'a> {
+impl HttpClient {
+    pub fn new(credentials: Credentials, serializer: Box<Serializer + Send + Sync>, hurl: Box<Hurl + Send + Sync>) -> HttpClient {
```

Comment:
Can be replaced with `Self`

Reply:
Refactoring the crate is out of scope for this PR. I agree, `-> HttpClient` should be refactored to `-> Self` in this case, but I'd be happy to do that in a separate PR.

Reply:
this is such a minor change that it doesn't warrant its own merge request

Reply:
There's lots of these "minor changes", which all together warrant a new PR.

Reply:
which is still no explanation for ignoring minor changes in code that you've touched anyways